src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1330,26 +1334,24 @@ DQUnpackDictGatherGQi::DQUnpackDictGatherGQi(Context::Ref ctx) {
 
 // This is a companion to DQLiftGatherAsymCW step. This pass runs if
 // the respective block (mainly, a head) was turned a function
-// (e.g. with FUNCALL_FOR_ALL) As in this case the HostGatherQuant
-// compile-time converts asymmetric MM to fp16, do the same thing here
-// Overall it's a combination of DQUnpackDictGather and HostGather
-// but we do unpack and gather in the runtime.
-HostGatherQuantAsymm::HostGatherQuantAsymm(Context::Ref ctx) {
+// (e.g. with FUNCALL_FOR_ALL)
+HostGatherQuantAsymm::HostGatherQuantAsymm(Context::Ref ctx, bool verify_only) {
     auto pids = opp::wrap_type<ov::op::v0::Parameter>();
     auto cvtids = opp::optional<ov::op::v0::Convert>({pids->output(0)});
 
-    auto qweight = opp::wrap_type<ov::op::v0::Parameter>();
-    auto qzerop = opp::wrap_type<ov::op::v0::Parameter>();
-    auto qcoeff = opp::wrap_type<ov::op::v0::Parameter>();
+    auto qweight = verify_only ? opp::wrap_type<ov::op::v0::Constant>() : opp::wrap_type<ov::op::v0::Parameter>();
+    auto qzerop = verify_only ? opp::wrap_type<ov::op::v0::Constant>() : opp::wrap_type<ov::op::v0::Parameter>();
+    auto qcoeff = verify_only ? opp::wrap_type<ov::op::v0::Constant>() : opp::wrap_type<ov::op::v0::Parameter>();
```

Comment:
what this `verify_only` is about?

Reply:
`verify_only` is used on the initial model (thus the Constants) to check if we have asymm quantization for the vocab. Without `verify_only` we do normal pass for quantized host gather. This flag is just to reuse the pattern. I can create a new pattern for that, to make it more readable